### ISSUE TITLE
Adjust Learning Rate Manually

### ISSUE THREAD

**Issue by pford221** [rstudio/keras#249]
Hi, 

I apologize if this has already been asked and answered, but I cannot find anywhere in the Keras for R documentation or in this repo how to change the learning rate between epochs manually without having to recompile. Keras for Python has `K.set_value(model.optimizer.lr, 1e-5)` .

 Is there a way to do this with R's `k_set_value()`?

**Comment by jjallaire** [comment]
you could do:

```r
k_set_value(model$optimizer$lr = 1e-5)
```

Or you could use use the learning rate scheduler callback: https://keras.rstudio.com/reference/callback_learning_rate_scheduler.html

**Comment by pford221** [comment]
Oy, that makes sense. Thanks, JJ. Yes, I'm aware of the `learning_rate_scheduler` but when I'm just trying to get a feel for a new model, I like to interactively fiddle with the learning rates. A more fundamental question, however, is that it appears as if the weights are not reset to random when running `compile`. That is, I can train a model for a certain number of epochs, then run `compile` again, with the expectation that this will reset the model weights to be randomly initialized, only to find that the loss metrics that are output to the console are where you'd expect them to be if we had picked up training from where we last left off in the prior epoch. So, does running `compile` not reset the weights?

**Comment by jjallaire** [comment]
I don't think running `compile()` resets the weights.

**Comment by pford221** [comment]
To  close the loop, I have found that in version 2.12 I am able to set the learning rate outside of a compilation step by running:

`k_set_value(model$optimizer$lr, 1e-5) `

This is slightly different than:
`k_set_value(model$optimizer = 1e-5)`
